### PR TITLE
DAOS-9820 test: make large datamover tests more configurable

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: datamover,-iosysadmin,-dm_posix_meta_entry
Skip-test: true
Skip-build: true

- Add `KeywordParameter` class that accepts a mapping of replaced values
- `data_mover_test_base`
  - Support ppn for various datamover utils
  - Use `KeywordParameter` for `posix_root`
- `large_dir` and `large_file`
  - Specify ppn instead of np
  - Dual-engine and move to HW Medium
  - Specify posix_root in the yaml
- `large_dir`
  - Remove duplicate variants in favor of the larger
  - Use EC object class by default, similar to `large_file`

Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>

### CODE REVIEW

**Comment by daltonbohning** [comment]
3 known test failures: https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-8662/10/tests
https://daosio.atlassian.net/browse/DAOS-10344

Looking at the log for `large_file`: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8662/10/artifact/Functional%20Hardware%20Medium/datamover/large_file.py/job.log
ppn is updated for IOR:
```
Updated param mpirun.ppn => 20
Updated param mpirun.np => None
...
mpirun ior -ppn 20
```
ppn is updated for `dcp`:
```
Updated param mpirun.ppn => 32
Updated param mpirun.np => None
...
mpirun dcp -ppn 32
```

**Comment by daltonbohning** [comment]
@phender This PR introduces the `KeywordParameter` as we discussed, though I wonder if a better name might be `MappedParameter` ?

**Review by shimizukko — APPROVED** [review]

File: src/tests/ftest/util/data_mover_test_base.py [hunk 1]
```diff
@@ -73,19 +76,31 @@ def __init__(self, *args, **kwargs):
...
+        self.dcp_np = None
+        self.dsync_np = None
+        self.dserialize_np = None
+        self.ddeserialize_np = None
+
+        # Root directory for POSIX paths
+        self.posix_root = KeywordParameter(None, mapping={
+            'self.workdir': self.workdir,
+            'self.tmp': self.tmp
+        })
```
Comment (phender) [inline src/tests/ftest/util/data_mover_test_base.py:103]: Minor cosmetic suggestion + adding the default:
```suggestion
        posix_root_map = {'self.workdir': self.workdir, 'self.tmp': self.tmp}
        self.posix_root = KeywordParameter(None, mapping=posix_root_map, default=self.tmp)
```

File: src/tests/ftest/util/data_mover_test_base.py [hunk 2]
```diff
@@ -114,23 +129,23 @@ def setUp(self):
...
+        self.dcp_np = self.params.get("np", "/run/dcp/*", self.processes)
+        self.dcp_ppn = self.params.get("ppn", "/run/dcp/*", self.ppn)
+        self.dsync_np = self.params.get("np", "/run/dsync/*", self.processes)
+        self.dsync_ppn = self.params.get("ppn", "/run/dsync/*", self.ppn)
+        self.dserialize_np = self.params.get("np", "/run/dserialize/*", self.processes)
+        self.dserialize_ppn = self.params.get("ppn", "/run/dserialize/*", self.ppn)
+        self.ddeserialize_np = self.params.get("np", "/run/ddeserialize/*", self.processes)
+        self.ddeserialize_ppn = self.params.get("ppn", "/run/ddeserialize/*", self.ppn)
+
+        self.posix_root.update(self.params.get("posix_root", "/run/datamover/*", self.tmp))
```
Comment (phender) [inline src/tests/ftest/util/data_mover_test_base.py:148]: Assign the default with creating the `KeywordParameter` object and then use it's `get_yaml_value()` method to read the test yaml:
```suggestion
        self.posix_root.get_yaml_value("posix_root", self, "/run/datamover/*")
```

**Review by phender — CHANGES_REQUESTED** [review]

**Comment by phender** [comment]
> @phender This PR introduces the `KeywordParameter` as we discussed, though I wonder if a better name might be `MappedParameter` ?

That works too, and its probably more descriptive.

**Comment by daltonbohning** [comment]
obj_large_posix and serial_large_posix are known failures:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-8662/13/tests/
https://daosio.atlassian.net/browse/DAOS-10344

**Comment by daltonbohning** [comment]
Ping @phender

**Review by dinghwah — APPROVED** [review]
LGTM

**Comment by daltonbohning** [comment]
> LGTM. Please address the 3 failures which might not related to the commit.

Thanks @dinghwah. The 3 failures are known: https://daosio.atlassian.net/browse/DAOS-10344

**Review by phender — APPROVED** [review]

**Pull request merged**